core/services/pipeline/runner.go

```diff
@@ -124,21 +135,30 @@ func NewRunner(orm ORM, btORM bridges.ORM, cfg Config, bridgeCfg BridgeConfig, l
 		httpClient:             httpClient,
 		unrestrictedHTTPClient: unrestrictedHTTPClient,
 	}
+
 	r.runReaperWorker = commonutils.NewSleeperTask(
 		commonutils.SleeperFuncTask(r.runReaper, "PipelineRunnerReaper"),
 	)
+
 	return r
 }
 
 // Start starts Runner.
-func (r *runner) Start(context.Context) error {
+func (r *runner) Start(ctx context.Context) error {
 	return r.StartOnce("PipelineRunner", func() error {
+		// starting cache can happen here for ORM
 		r.wgDone.Add(1)
 		go r.scheduleUnfinishedRuns()
 		if r.config.ReaperInterval() != time.Duration(0) {
 			r.wgDone.Add(1)
 			go r.runReaperLoop()
 		}
+
+		cache, isCache := r.btORM.(*bridges.Cache)
```

Comment:
When would it not be a cache?

Could also check if it implements `Start` to make it more generally applicable